src/system/tests/TestSystemTimer.cpp

```diff
@@ -206,6 +207,72 @@ static void CheckStarvation(nlTestSuite * inSuite, void * aContext)
     ServiceEvents(lSys, sleepTime);
 }
 
+void HandleStartTimer(Layer * inetLayer, void * aState, Error aError)
+{
+    TestContext & lContext = *static_cast<TestContext *>(aState);
+    NL_TEST_ASSERT(lContext.mTestSuite, true);
+    sStartTimerHandled = true;
+}
+
+static void CheckStartTimer(nlTestSuite * inSuite, void * aContext)
+{
+    uint32_t timeout_100ms = 100;
+
+    TestContext & lContext = *static_cast<TestContext *>(aContext);
+    Layer & lSys           = *lContext.mLayer;
+
+    sStartTimerHandled = false;
+
+    Error err = lSys.StartTimer(timeout_100ms, HandleStartTimer, aContext);
+    NL_TEST_ASSERT(lContext.mTestSuite, err == CHIP_SYSTEM_NO_ERROR);
+
+    while (!sStartTimerHandled)
+    {
+        struct timeval sleepTime;
+        sleepTime.tv_sec  = 0;
+        sleepTime.tv_usec = 1000; // 1 ms tick
+        ServiceEvents(lSys, sleepTime);
+    }
+
+    NL_TEST_ASSERT(lContext.mTestSuite, sStartTimerHandled == true);
+    lSys.CancelTimer(HandleStartTimer, aContext);
```

Comment:
Shouldn't this have a test to confirm the negative state? What if we pass in no context?